File: app/src/main/java/it/gruppoinfor/home2work/fragments/ConfigurationFragmentComplete.java
package it.gruppoinfor.home2work.fragments;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.firebase.iid.FirebaseInstanceId;
import com.stepstone.stepper.BlockingStep;
import com.stepstone.stepper.StepperLayout;
import com.stepstone.stepper.VerificationError;

import butterknife.ButterKnife;
import it.gruppoinfor.home2work.R;
import it.gruppoinfor.home2workapi.Client;
import it.gruppoinfor.home2workapi.model.User;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ConfigurationFragmentComplete extends Fragment implements BlockingStep {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_conf_completed, container, false);
        ButterKnife.bind(this, root);
        return root;
    }

    @Override
    public VerificationError verifyStep() {
        return null;
    }

    @Override
    public void onSelected() {

    }

    @Override
    public void onError(@NonNull VerificationError error) {

    }

    @Override
    public void onNextClicked(StepperLayout.OnNextClickedCallback callback) {

    }

    @Override
    public void onCompleteClicked(final StepperLayout.OnCompleteClickedCallback callback) {
        callback.getStepperLayout().showProgress("Attendi..");

        Client.getAPI().updateUser(Client.getSignedUser()).enqueue(new Callback<User>() {
            @Override
            public void onResponse(Call<User> call, Response<User> response) {
                callback.getStepperLayout().hideProgress();
                if(response.code()==200){
                    callback.complete();
                }
            }

            @Override
            public void onFailure(Call<User> call, Throwable t) {
                callback.getStepperLayout().hideProgress();
            }
        });

    }

    @Override
    public void onBackClicked(final StepperLayout.OnBackClickedCallback callback) {
        callback.goToPrevStep();
    }
}

File: App/src/main/java/it/gruppoinfor/home2work/configuration/ConfigurationFragmentComplete.kt
package it.gruppoinfor.home2work.configuration

import android.annotation.SuppressLint
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.stepstone.stepper.BlockingStep
import com.stepstone.stepper.StepperLayout
import com.stepstone.stepper.VerificationError
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers
import it.gruppoinfor.home2work.R
import it.gruppoinfor.home2work.api.HomeToWorkClient

class ConfigurationFragmentComplete : Fragment(), BlockingStep {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        return inflater.inflate(R.layout.fragment_conf_completed, container, false)
    }

    override fun verifyStep(): VerificationError? {

        return null
    }

    override fun onSelected() {

    }

    override fun onError(error: VerificationError) {

    }

    override fun onNextClicked(callback: StepperLayout.OnNextClickedCallback) {

    }

    @SuppressLint("CheckResult")
    override fun onCompleteClicked(callback: StepperLayout.OnCompleteClickedCallback) {

        callback.stepperLayout.showProgress(getString(R.string.activity_configuration_wait))

        HomeToWorkClient.user?.configured = true

        HomeToWorkClient.getUserService().edit(HomeToWorkClient.user!!)
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .onErrorReturn {
                    null
                }
                .subscribe({
                    callback.complete()
                }, {
                    callback.stepperLayout.hideProgress()
                })


    }

    override fun onBackClicked(callback: StepperLayout.OnBackClickedCallback) {

        callback.goToPrevStep()

    }
}